Compressonator/Applications/_Plugins/CCMP_Encode/HPC/Compute_CPU_HPC.cpp

```diff
@@ -43,8 +43,8 @@ unsigned int    _stdcall ProcEncode(void* param)
 {
    ThreadParam *tp = (ThreadParam*)param;
 
-   tp->run = false;
    //printf("Thead Active [%4x]\n",std::this_thread::get_id());
+   std::this_thread::sleep_for(0ms);
 
    while (tp->exit == false)
    {
```

Comment:
FYI: This “patch” does not fix anything. Multi-threading is NOT about timing. The “solution” you have presented and adopted clearly proves that you probably have no idea what you are doing or talking about.

If you want to *synchronize* the launch time of threads then you should use a [barrier](https://en.wikipedia.org/wiki/Barrier_(computer_science)) before `CompressBlock()`. However, this does not guarantee synchronous execution of all threads after releasing the barrier either. So, better yet, employ a traversing mutex per block and thread. Additionally, all of these `std::this_thread::sleep_for(0)` calls are pure nonsense.

You also do not need `ThreadParam.run`. You can check a thread’s execution state by calling [`std::thread::joinable()`](https://en.cppreference.com/w/cpp/thread/thread/joinable).

Reply:
I will agree that line 47 `std::this_thread::sleep_for(0ms);` is not necessary and maybe it should be removed, it was added in `yield()` sense.
I will also agree that in this patch my pull request is not formal and quickly pushed.
I will disagree that this patch doesn't fix anything. Please read again the pull request and look the code please. Removing `tp->run=false` at line 46 fix a race condition with line ~179 `m_EncodeParameterStorage[threadIndex].run = true;`. If the later happen before the former the block won't be encoded. This bug surfaced to two systems of mine after a Windows update (iirc).
Also I want to inform you that for me your attitude was a little aggressive, please I have no much heart to tolerate that.
Have a nice day.